examples/data_view/column_data_model.py

```diff
@@ -0,0 +1,232 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from abc import abstractmethod
+
+from traits.api import (
+    ABCHasStrictTraits, ComparisonMode, Event, HasTraits, Instance,
+    List, Str, observe
+)
+from traits.trait_base import xgetattr, xsetattr
```

Comment:
I think that this will eventually become a proper data model in the library, rather than an example.  It is an example for now because it demonstrates a capability, but the general case it represents is one we want to support generically.

Reply:
My main concern is that it will be quite some time before this model becomes a built-in model, and then in the meantime, people need the feature and they start copying from this example, and whatever anti-patterns here are copied all over the place. 

If this model is not ready to be copied by others, I'd prefer we defer making this visible. We can have a separate branch or tag where we stash this work.

Reply:
(I'd still prefer we defer making not-yet-ready example visible). But we must merge this to master, we should at the very least have a very visible comment here to stop people from copying this example.

Reply:
Secondary concern: I _want_ people to be using `xgetattr` and `xsetattr` instead of rolling their own, badly.  Or worse, using `eval`.

Reply:
And I guess, there is a secondary thing, that we are getting bogged down in perfecting things that weren't meant to be perfect.  This example was included so that _reviewers_ could understand how the code worked and have concrete examples of how to do things and the space of stuff that we want to support.

Reply:
Understood. The context for these examples is a bit different here. For developers (not us) using pyface, it is typically assumed that code in examples are demonstrations of how things should be or can be written.  And it is fair to assume that people can and will happily copy example code and then adapt from it.

I am not trying to aim for perfection here, but I am hoping that examples should be good enough to be copied and adapted far and wide. People browsing the examples folder won't have the context that this example is kind-of-a work-in-progress, they will assume that this is good enough for them to copy. This is not just about `xgetattr` actually, but the whole example in general.

I agree and appreciate that these examples are really valuable for the purpose of demonstrating to the reviewers that the code works. Whether it needs to be merged to master is another issue. I would have preferred them not to be merged, but I am okay with merging as long as we can manage the expectation from other developers that these examples are not ready to be copied yet. A comment at the top would do that.

Reply:
I think at this point there are no other developers, so the issue is moot.

Reply:
What I am concerned is probably a bit far fetched:

Say we merged this to master, expecting this will be cleaned up later, then a lot of other things more important and more urgent came along, eventually we get to the point where we will be releasing the data view feature without the column data example being cleaned and converted to a built-in model.  Then users of pyface want to build a column based view, then they discover this example because it is visible on master along with the released base classes and all. So they copy them to their own code base. It is in the example folder so it ought to be good enough right? 

All I am asking is a big fat comment in the example that says this is still a work-in-progress not ready to be adopted widely yet... just in case. I hope this is not a big ask.

Reply:
I think it's fair to say that 7.1 doesn't get released without the data view being solid.  There isn't much more that is going important for the next minor release.